R/inverse_interpolation.R

```diff
@@ -21,3 +21,43 @@ inverse_interpolation <- function (x, y, ypred, omega) {
     return(xpred)
 }
 
+
+#' Local inverse interpolation with bandwidth selected via k-fold cross-validation.
+#'
+#' @param x The matrix of "training" values of x, with one row per training set.
+#' @param y The matrix of "training" values of y corresponding to x.
+#' @param ypred The matrix of "test" values of y for which a value of x will be predicted.
+#' @param seed The random seed used in the k-fold cross-validation.
+#' @param min_omega The minimum bandwidth attempted. 
+#' @param max_omega The initial maximum bandwidth attempted. 
+#' @param k The number of folds to use in cross-validation.
+#' @export
+#' @return List of `omega_min`, the omega that minimizes the cross-validation error,
+#'                 `ypred` the matrix of predicted x values for the optimal
+#'                 omega, with dimensions (nrow(ypred), ncol(x)), and
+#'                 `boundary_value` a boolean indicating if `omega_min` is on
+#'                 the boundary of the range of omegas examined.
+automatic_inverse_interpolation <- function (x, y, ypred, seed, min_omega, max_omega=NA, k=5) {
+    max_omega <- 50 * min_omega
+    omegas <- seq(min_omega, max_omega, length.out=100)
+    idxs <- 1:length(x)
+    set.seed(seed)
+    shuffled_idxs <- sample(idxs, length(idxs), replace=FALSE)
+    folds <- split(shuffled_idxs, cut(shuffled_idxs, breaks=k))
```

Comment:
nice! I did not know about `split()`!

Reply:
But, oh, this is not what you mean.  Cut divides by *values*.  Notice this always returns the same groups:
```
shuffled_idxs <- sample.int(10)
k <- 2
split(shuffled_idxs, cut(shuffled_idxs, breaks=k))
```

I think you mean
```
    folds <- split(shuffled_idxs, cut(seq_along(shuffled_idxs), breaks=k)) 
```